api/src/main/java/org/openmrs/module/sync/api/impl/SyncIngestServiceImpl.java

```diff
@@ -565,7 +570,22 @@ else if (!alreadyExists && isDelete) {
                         SyncUtil.setProperty(o, xmlFieldName, null);
                     }
                     else {
-                        SyncUtil.setProperty(o, field, xmlFieldName, xmlFieldValue, xmlFieldType);
+                        if (o instanceof Obs && xmlFieldName.equals("referenceRange")) {
+                            // If an obs comes through with a reference range, and that reference range is not found, ignore that error.
+                            try {
+                                SyncUtil.setProperty(o, field, xmlFieldName, xmlFieldValue, xmlFieldType);
+                            }
+                            catch (Exception e) {
+                                log.warn("Not yet able to set reference range of " + xmlFieldValue + " on obs");
+                            }
+                        }
+                        else {
+                            // If reference range comes through, keep track of what obs that is associated with so that obs can be updated
+                            if (className.equals("org.openmrs.ObsReferenceRange") && xmlFieldName.equals("obs")) {
+                                obsToUpdateWithReferenceRange = xmlFieldValue;
+                            }
+                            SyncUtil.setProperty(o, field, xmlFieldName, xmlFieldValue, xmlFieldType);

```

Comment:
... ie this line is the "normal path" and it kind of gets buried, maybe worth a one-line comment here as well